pkg/ebpf/tracer.go

```diff
@@ -787,3 +787,23 @@ func (t *Tracer) determineConnectionDirection(conn *network.ConnectionStats) net
 
 	return network.OUTGOING
 }
+
+func (t *Tracer) getProbeProgramIDs() (map[string]uint32, error) {
+	fds := make(map[string]uint32, 0)
+	for _, p := range t.m.Probes {
+		if !p.Enabled {
+			continue
+		}
+		prog := p.Program()
+		if prog == nil {
+			fmt.Printf("unable to find program for %s\n", p.Section)
```

Comment:
Since `ebpf.Manager` is defined ahead-of-time, and then `Init` is called with the ELF program. If you called this function before `Init`, then no programs would be loaded yet.